# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1467746104 25200
#      Tue Jul 05 12:15:04 2016 -0700
# Node ID 26c0e65e6b7751ea4a45a4492df7eb2b68ac0bc7
# Parent  79e7a6fc524d1d3ee8285828e7059c0e3e16208b
configure: clean up warning messages when building without GraphicsMagick

* configure.ac: Add a warning message when pkg-config fails to find
GraphicsMagick.  Reword all GraphicsMagick warning messages to mention all
affected functions.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1146,17 +1146,17 @@ AC_SUBST(PORTAUDIO_LIBS)
 
 check_magick=yes
 use_magick=no
 AC_ARG_WITH([magick],
   [AS_HELP_STRING([--with-magick=LIB],
     [select library to use for image I/O (options: GraphicsMagick(default) or ImageMagick)])],
   [if test x"$withval" = xno; then
      check_magick=no
-     warn_magick_disabled="--without-magick specified.  The imread function for reading images will not be fully functional."
+     warn_magick_disabled="--without-magick specified.  The imread, imwrite, and imfinfo functions for reading and writing image files will not be fully functional."
      OCTAVE_CONFIGURE_WARNING([warn_magick_disabled])
    else
      magick="$withval"
    fi], [magick="GraphicsMagick"])
 
 if test $check_magick = yes; then
 
   MAGICK_CPPFLAGS=
@@ -1167,17 +1167,17 @@ if test $check_magick = yes; then
     ## Make sure we only get -I, -L, and -l flags.  Some Graphics/ImageMagick++
     ## packages add extra flags that are useful when building
     ## Graphics/ImageMagick++ extentions.  These extra flags break the
     ## Octave build.
     MAGICK_CPPFLAGS=`$PKG_CONFIG --cflags-only-I $magick++`
     MAGICK_LDFLAGS=`$PKG_CONFIG --libs-only-L $magick++`
     MAGICK_LIBS=`$PKG_CONFIG --libs-only-l $magick++`
 
-    warn_magick="$magick++ library fails tests.  The imread function for reading image files will not be fully functional."
+    warn_magick="$magick++ library fails tests.  The imread, imwrite, and imfinfo functions for reading and writing image files will not be fully functional."
 
     save_CPPFLAGS="$CPPFLAGS"
     save_LIBS="$LIBS"
     CPPFLAGS="$MAGICK_CPPFLAGS $CPPFLAGS"
     LIBS="$MAGICK_LDFLAGS $MAGICK_LIBS $LIBS"
     AC_LANG_PUSH(C++)
     AC_CHECK_HEADER([Magick++.h], [
       AC_CACHE_CHECK([for Magick::ColorRGB in Magick++.h],
@@ -1196,18 +1196,20 @@ if test $check_magick = yes; then
       fi
     ])
     AC_LANG_POP(C++)
     CPPFLAGS="$save_CPPFLAGS"
     LIBS="$save_LIBS"
 
     AC_CHECK_FUNCS([setlocale], [],
       [use_magick=no
-       warn_magick="$magick++ requires setlocale function.  The imread function for reading image files will not be fully functional."])
-  ])
+       warn_magick="$magick++ requires setlocale function.  The imread, imwrite, and imfinfo functions for reading and writing image files will not be fully functional."])
+  ],
+  [use_magick=no
+   warn_magick="$magick++ library not found.  The imread, imwrite, and imfinfo functions for reading and writing image files will not be fully functional."])
 fi
 
 if test $use_magick = yes; then
   AC_DEFINE(HAVE_MAGICK, 1,
     [Define to 1 if Graphics/ImageMagick++ is available.])
 else
   if test -n "$warn_magick"; then
     OCTAVE_CONFIGURE_WARNING([warn_magick])
